### ISSUE TITLE
x86 code generator fails to handle floating-point to unsigned integer conversion opcodes

### ISSUE THREAD

**Issue by Leonardo2718** [eclipse/omr#2375]
Currently, on x86, a floating-point to unsigned integer conversion opcode, such as `f2iu`, causes an assert failure in `TR::TreeEvaluator::f2iEvaluator`:

https://github.com/eclipse/omr/blob/c96856abd98471b92f0415e1207d990c1301a7af/compiler/x/codegen/FPTreeEvaluator.cpp#L1371-L1405

As far as I can tell, those x86 instructions are signed. So just copying the cases for the signed opcodes likely won't work.

@andrewcraik @0xdaryl @mstoodle and @aviansie-ben, any thoughts?

**Comment by andrewcraik** [comment]
I don't believe that x86 has a built-in instruction for these operations and copying won't work. There would need to be a sequence to generate something useful for the particular cases, but it seems like a rarely used feature given the trouble I'm having figuring out what the definitions of f2iu and d2iu mean in terms of how the conversion would work.

For the int case you might be able to do the long signed convert and then l2iu? It does depend on what is expected to be produced for negative floating point values....

**Comment by aviansie-ben** [comment]
As far as I know, there's no instructions in x86-64 for converting floating-point values to unsigned integers directly. The `f2iu` and `d2iu` instructions should be possible to implement by just performing the conversion to a long and just using the bottom 32 bits of the result. The `f2lu` and `d2lu` instructions are problematic though and I don't know of any easy way of accomplishing that.

**Comment by mstoodle** [comment]
Shouldn't there also be an overflow effect captured in there too (i.e. not "just using the bottom 32 bits of the result") ?
